weatherApp/Features/WeatherList/UI/HomeViewController.swift

```diff
@@ -53,29 +56,23 @@ class HomeViewController: UIViewController {
     
     //MARK: - TableView Data
     
-    @objc private func setupData() {
-        let locationIds = Cities.allCases.map { $0.rawValue }
+    @objc private func bindViewModel() {
+        if currentWeatherListPresenter == nil {
+            currentWeatherListPresenter = CurrentWeatherListPresenter()
+        }
```

Comment:
You can remove this:
```
if currentWeatherListPresenter == nil {
            currentWeatherListPresenter = CurrentWeatherListPresenter()
}
```

You already initialized presenter inside init of a view controller.